src/PrettyPrinting.jl

```diff
@@ -1591,23 +1591,89 @@ function _write_line(io::IOCustom, str::AbstractString)
     spaceleft = c - ind - io.printed
   end
   #@show spaceleft
-  firstlen = min(spaceleft, length(str))
-  firststr = str[1:firstlen]
-  if io.lowercasefirst
-    written += write(io.io, lowercasefirst(firststr))
-    io.lowercasefirst = false
+  _graphemes = Base.Unicode.graphemes(str)
+  firstlen = min(spaceleft, length(_graphemes))
+  # make an iterator over valid indices
+  firstiter = Base.Iterators.take(_graphemes, firstlen)
+  restiter = Base.Iterators.drop(_graphemes, firstlen)
+  firststr = join(firstiter)
```

Comment:
Could it be that an uppercase character has a different width than a lower case character? I hope not, but maybe it is safer to todo
```suggestion
  firststr = join(firstiter)
  if io.lowercasefirst
    firststr = lowercasefirst(firststr)
    io.lowercasefirst = false
  end
```
and then all lowercase treatment afterwards can go away